Automattic/jetpack
# Licensing: exclude revoked licenses from unattached license filter

Fixes https://github.com/Automattic/jetpack-roadmap/issues/2010

This will prevent users with only revoked unattached licenses from being redirected to the license activation page after connecting their site to Jetpack.

## Proposed changes:

* Updated the `get_user_licenses` method in the `Licensing` class to exclude licenses marked as revoked (revoked_at !== null) when filtering unattached licenses.


### Other information:


## Jetpack product discussion


p1HpG7-uUZ-p2#comment-74755

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






- Spin up a Jurassic Ninja site with Jetpack VaultPress Backup using this branch
- Ensure your user has no licenses available, and has at least one revoked license
- Navigate to Jetpack > VaultPress Backup
- Click on `Get VaultPress Backup`
- Approve the Jetpack Connection
- Validate you are redirected to the checkout page instead of the license activation page

https://github.com/user-attachments/assets/ebe7ce8a-2d4e-44f3-b113-7b6cfddc5a93

If you want to reproduce the bug reported, you can follow the steps described in https://github.com/Automattic/jetpack-roadmap/issues/2010.
